test/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountControllerSpec.scala

```diff
@@ -135,8 +136,28 @@ class AccessAccountControllerSpec
         test(Some(HTSSession.empty))
       }
 
+      "redirect to protect tax if nino from user session doesn't have corresponding pt enrolment" in {
+        mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval and noPersonalTaxEnrolment)
+        mockSessionStoreGet(Right(Some(HTSSession.empty.copy(attemptedAccountHolderPageURL = Some("abc")))))
+        mockEnrolmentCheck()(Right(EnrolmentStatus.Enrolled(true)))
+
+        val result = doRequest()
+        status(result) shouldBe 303
+        redirectLocation(result) shouldBe Some(appConfig.protectTaxInfoUrl)
+      }
+
+      "redirect to protect tax if nino from user session doesn't match nano from PT enrolment" in {
+        mockAuthWithNINORetrievalWithSuccess(AuthWithCL200)(mockedNINORetrieval and enrolmentsWithNoMatchingNino)
+        mockSessionStoreGet(Right(Some(HTSSession.empty.copy(attemptedAccountHolderPageURL = Some("abc")))))
```

Comment:
I'm guessing that we probably don't need these two mocks as the request does not get that far